### ISSUE TITLE
Restic: restore from old backups

### ISSUE THREAD

**Issue by JohnStrunk** [backube/volsync#7]
**Describe the feature you'd like to have.**
Currently, the restic mover only supports restoring the latest backup even though we have retention policies that keep multiple older backups.

Something like:
```yaml
spec:
  restic:
    restoreAsOf: "2021-07-14T12:56:27Z"
    previous: 2
```
Both fields would be optional, and if not specified would default to "the current time" and 0, respectively.

The "as-of" time refers to the backup that is the most recent as of that time. For example, if the as-of time is yesterday at noon, it would refer to the most recent backup still in the repo that is older than yesterday at noon.

The "previous" field can be used to reference older backups in sequence. For example, `previous: 0` would be the most recent backup, and `previous: 1` would be the backup prior to the most recent.

**What is the value to the end user? (why is it a priority?)**
Today, users can only restore from the most recent backup using Scribe. If they want to access an older backup, they must directly access the restic repo via the restic cli, and they are on their own for getting that data back into a PVC.

**How will we know we have a good solution? (acceptance criteria)**

**Additional context**
Restic normally requires specifying the "snapshot id" of the backup in order to restore it. The above proposal is designed to avoid having to deal directly with the IDs which would require some sort of interface to list them and have the user then specify the desired one.

The scribe cli could also be useful here to give access to the restic repo by getting the credentials from the cluster and invoking restic on the user's behalf.

**Comment by osilkin98** [comment]
When `previous` is bigger than the number of snapshots, should it select the oldest available snapshot or nothing?

**Comment by JohnStrunk** [comment]
Fixed in #39
